airflow/triggers/external_task.py

```diff
@@ -38,25 +39,33 @@ class TaskStateTrigger(BaseTrigger):
     :param task_id: The task_id that contains the task you want to
         wait for. If ``None`` (default value) the sensor waits for the DAG
     :param states: allowed states, default is ``['success']``
```

Comment:
No such default is currently set for this parameter.

Reply:
Currently, for the `states` parameter, as it is in the description, we have set it to `["success"]` by default.
list(allowed_states) if allowed_states else [TaskInstanceState.SUCCESS.value].

Reply:
This default is set in the sensor though, in the trigger itself this behavior is not implemented. If someone now wanted to use the trigger seperately, the documentation for it is incorrect.